lib/resources/postgres.rb

```diff
@@ -18,27 +19,22 @@ def initialize
         # Debian allows multiple versions of postgresql to be
         # installed as well as multiple "clusters" to be configured.
         #
-        version = version_from_dir('/etc/postgresql')
-        cluster = cluster_from_dir("/etc/postgresql/#{version}")
-        @conf_dir = "/etc/postgresql/#{version}/#{cluster}"
-        @data_dir = "/var/lib/postgresql/#{version}/#{cluster}"
+        @version = version_from_psql.to_s
+        @version = version_from_dir('/etc/postgresql') if @version.nil?
+        @cluster = cluster_from_dir("/etc/postgresql/#{@version}")
+        @conf_dir = "/etc/postgresql/#{@version}/#{@cluster}"
+        @data_dir = "/var/lib/postgresql/#{@version}/#{@cluster}"
       elsif os.redhat?
-        #
-        # /var/lib/pgsql/data is the default data directory on RHEL6
-        # and RHEL7. However, PR #824 explicitly added version-based
-        # directories. Thus, we call #version_from_dir unless it looks
-        # like we are using unversioned directories.
-        #
-        # TODO(ssd): This has the potential to be noisy because of the
-        # warning in version_from_dir. We should determine which case
-        # is more common and only warn in the less common case.
-        #
-        version = if inspec.directory('/var/lib/pgsql/data').exist?
-                    warn 'Found /var/lib/pgsql/data. Assuming postgresql install uses un-versioned directories.'
-                    nil
-                  else
-                    version_from_dir('/var/lib/pgsql/')
-                  end
+        @version = version_from_psql.to_s
```

Comment:
well we wouldn't run version_from_dir on /var/lib/pgsql/data, it is just checking if that dir exists. If it does, it was assumed that means the version structure isn't the standard of /var/lib/pgsql/'<version>'/data thus we could not find the version number via the directories given '<version>' wasn't there, so it was just let unset. 

Using the new function we will always get a version so this logic really don't need to be there any more, so we can shorten it. The else in this cases was to say if 'var/lib/pgsql/data' doesn't exist - we assume that we can run version from dir on '/var/lib/pgsql' and assume to find the ver dir waiting for us. I will see about cleaning up the logic.